src/contexts/providers.context.tsx

```diff
@@ -133,6 +165,27 @@ const ProvidersProvider: FC = (props) => {
     }
   }, [connectedProvider]);
 
+  const estimateGasPrice = useCallback(
+    ({ chain, gasUnits }: EstimateGasPriceParams): Promise<BigNumber | undefined> => {
+      if (chain.provider === undefined) {
+        throw new Error("Provider is not available");
+      }
+
+      return chain.provider.getFeeData().then((feeData) => {
+        if (feeData.maxFeePerGas !== null) {
+          return gasUnits.mul(feeData.maxFeePerGas);
+        }
+
+        if (feeData.gasPrice !== null) {
+          return gasUnits.mul(feeData.gasPrice);
+        }
+
+        return undefined;
```

Comment:
Should'n we throw instead of returning `undefined`? If I understood well, we must either get `maxFeePerGas` or `gasPrice` so both being `null` looks like an error.

Reply:
Right! Fixed